### PR TITLE
Tpetra:  deprecate node args (phase two):  remove node args in map

### PR DESCRIPTION
**Pull request by kddevin**
@trilinos/tpetra @trilinos/xpetra @trilinos/muelu @trilinos/stokhos 

## Description
Tpetra is deprecating use of Node as an argument in functions.  
This PR removes the Node argument from map creation, and removes other interfaces that take a Node argument.
Still remaining:  removal of Node from clone functions and TSQR.

## Motivation and Context
Node is made obsolete by Kokkos::ExecutionSpace and Kokkos::MemorySpace.

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to #4839 
* Part of 
* Composed of 

## How Has This Been Tested?

Builds of Tpetra with and without deprecated code.
Builds with MueLu, Stokhos, Nox, Thyra, Anasazi and Belos, with and without deprecated code.
See #4839 for details.

## Checklist

- [ ] All new and existing tests passed. 
SOME TESTS FAIL, BUT THEY ALSO FAIL IN DEVELOP BRANCH.
- [ ] No new compiler warnings were introduced.  
DEPRECATION WARNINGS ARE INTRODUCED
- [ ] These changes break backwards compatibility.
THAT'S THE POINT.

### CODE REVIEW

File: packages/tpetra/core/test/Map/Map_UnitTests.cpp [hunk 1]
```diff
@@ -426,7 +426,7 @@ namespace {
     RCP<N2> n2 (new N2);
 
     // create a contiguous uniform distributed map with numLocal entries per node
-    RCP<const Map1> map1 = createUniformContigMapWithNode<LO, GO, N1> (numGlobal, comm, n1);
+    RCP<const Map1> map1 = createUniformContigMapWithNode<LO, GO, N1> (numGlobal, comm);
```
Comment (lucbv) [inline packages/tpetra/core/test/Map/Map_UnitTests.cpp]: @kddevin should a test using the deprecated function signature be kept to ensure it works until the deprecated function signature is actually removed from the code?
We are having a similar discussion for some MueLu deprecation...

File: packages/tpetra/core/test/ImportExport2/ImportExport2_UnitTests.cpp [hunk 2]
```diff
@@ -2026,7 +2026,7 @@ TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL( Import_Util, UnpackAndCombineWithOwningPIDs,
   RCP<const MapType> MapSource, MapTarget;
 
   const bool debug = ::Tpetra::Details::Behavior::debug ();
-  const bool verbose = ::Tpetra::Details::Behavior::verbose ();
+  verbose = ::Tpetra::Details::Behavior::verbose ();
```
Comment (mhoemmen) [inline packages/tpetra/core/test/ImportExport2/ImportExport2_UnitTests.cpp]: This doesn't quite make sense -- did the unit test capture a `verbose` variable from outside?

File: packages/tpetra/core/test/Map/Map_UnitTests.cpp (hunk 1 above)
Comment (mhoemmen) [reply packages/tpetra/core/test/Map/Map_UnitTests.cpp]: I don't think we're obligated to keep supporting deprecated functions, but it certainly wouldn't hurt to leave a test in place, as long as it doesn't generate deprecated warnings if deprecated code is disabled.

Comment (kddevin) [reply packages/tpetra/core/test/Map/Map_UnitTests.cpp]: I admit I was not consistent.  In some places, I kept the old interface (e.g., https://github.com/trilinos/Trilinos/blob/e000e39a04969848a2e462def7dbde9e83f822ff/packages/tpetra/core/test/TpetraUtils_MatrixGenerator.hpp#L226 and https://github.com/trilinos/Trilinos/blob/e000e39a04969848a2e462def7dbde9e83f822ff/packages/tpetra/core/test/TpetraUtils_MatrixGenerator.hpp#L176 ).  In other places, I just removed calls to the old interface.

File: packages/tpetra/core/test/ImportExport2/ImportExport2_UnitTests.cpp (hunk 2 above)
Comment (kddevin) [reply packages/tpetra/core/test/ImportExport2/ImportExport2_UnitTests.cpp]: Removes a shadow warning from https://github.com/trilinos/Trilinos/blob/e000e39a04969848a2e462def7dbde9e83f822ff/packages/tpetra/core/test/ImportExport2/ImportExport2_UnitTests.cpp#L117

Comment (mhoemmen) [reply packages/tpetra/core/test/ImportExport2/ImportExport2_UnitTests.cpp]: @kddevin Thanks for explaining!  I would have somewhat preferred getting rid of the outer `verbose` variable, but the way you did it is fine.  Thanks!

**Comment by etphipp** [comment]
This generates build errors if Tpetra deprecated code is enabled:
```
/home/etphipp/Trilinos/Trilinos/packages/xpetra/src/Map/Xpetra_TpetraMap.hpp:643:55: error: 
      missing default argument on parameter
               const Teuchos::RCP< Node > & /* node */) 
                                                      ^
```
Looking at the code, you have a function argument with no default value after an argument with a default value:
```
#ifdef TPETRA_ENABLE_DEPRECATED_CODE
    TPETRA_DEPRECATED
    TpetraMap (global_size_t numGlobalElements,
               GlobalOrdinal indexBase,
               const Teuchos::RCP< const Teuchos::Comm< int > > &comm,
               LocalGlobal lg=GloballyDistributed,
               const Teuchos::RCP< Node > & /* node */)
      : TpetraMap(numGlobalElements, indexBase, comm, lg)
    {}
#endif // TPETRA_ENABLE_DEPRECATED_CODE
```
Can the default value for `lg` just be removed?

**Comment by kddevin** [comment]
Yes, "=GloballyDistributed" should be removed.  I will do it.  Sorry we missed that one (and that PR testing did not catch the problem).

**Pull request merged**